setup.py

```diff
@@ -18,67 +18,107 @@
 ##############################################################################
 """Setup for zope.app.apidoc package
 
-$Id$
 """
 import os
 from setuptools import setup, find_packages
 
 def read(*rnames):
-    return open(os.path.join(os.path.dirname(__file__), *rnames)).read()
+    with open(os.path.join(os.path.dirname(__file__), *rnames)) as f:
+        return f.read()
+
+tests_require = [
+    'zope.app.authentication >= 4.0.0',
+    'zope.app.folder >= 4.0.0',
+    'zope.app.http >= 4.0.1',
+    'zope.app.principalannotation >= 4.0.0',
+    'zope.app.rotterdam >= 4.0.0',
+    'zope.app.wsgi >= 4.1.0',
+    'zope.applicationcontrol >= 4.0.0',
+
+    'zope.browserpage >= 4.1.0',
+    'zope.login',
+    'zope.principalannotation',
+    'zope.securitypolicy',
+    'zope.testing',
+    'zope.testrunner',
+
+    # If we wanted to depend on lxml, it can parse the results quite
+    # a bit faster.
+
+    # Things we don't use or configure, but which are
+    # picked up indirectly by other packages and
+    # need to be loaded to avoid errors running the
+    # full static export.
+    'zope.app.component[test]',
+    'zope.app.form[test]', # zc.sourcefactory
+    'zope.app.schema[test]',
+]
+
+static_requires = tests_require
```

Comment:
I had it that way at one point, but then it turned out that they were identical...the static export loads the `ftesting.zcml` from this package, and it uses `zope.testbrowser`, so right now they pretty much have to be identical. In the future, when what it loads is configurable, it may make more sense to change them up.